### PR TITLE
Add delete option to attachments

### PR DESCRIPTION
**Pull request by rdjuric**
### Details

We change our `shouldShow` condition for the delete option to be a new function `canDeleteReportAction`. It is similar to our canEditReportAction, but don't check if the message is an attachment.

### Fixed Issues

#3775 

### Tests

1. Open any chat
2. Send an attachment 
3. The delete option shows on the menu
4. Clicking on the delete option correctly deletes the attachment
### QA Steps

1. Open any chat
2. Send an attachment 
3. The delete option shows on the menu
4. Clicking on the delete option correctly deletes the attachment

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/53711423/123856564-a283cb00-d8f7-11eb-9f73-f0e918e59b64.mov

#### Mobile Web

https://user-images.githubusercontent.com/53711423/123856665-bdeed600-d8f7-11eb-9d5a-7e11a9e050a7.mp4

#### Desktop

https://user-images.githubusercontent.com/53711423/123856726-cf37e280-d8f7-11eb-9764-1da0b528d5d2.mov

#### iOS

https://user-images.githubusercontent.com/53711423/123856813-e971c080-d8f7-11eb-9fa5-a29d8cb4a85a.mp4

#### Android

https://user-images.githubusercontent.com/53711423/123858297-b8928b00-d8f9-11eb-8085-cf9bf950215d.mp4

### CODE REVIEW

**Comment by rdjuric** [comment]
Just a note: while #3767 isn't fixed, we have to use the context menu from the side of the message to test this on the web/desktop.

File: src/libs/reportUtils.js [hunk 1]
```diff
@@ -46,7 +46,8 @@ function sortReportsByLastVisited(reports) {
 }
 
 /**
- * Can only edit if it's a ADDCOMMENT, the author is this user and it's not a optimistic response.
+ * Can only edit if it's a ADDCOMMENT that is not an attachment,
```
Comment (luacmartins) [inline src/libs/reportUtils.js:49]:
```suggestion
 * Can only edit if it's an ADDCOMMENT that is not an attachment,
```

File: src/libs/reportUtils.js [hunk 2]
```diff
@@ -46,7 +46,8 @@ function sortReportsByLastVisited(reports) {
...
+ * the author is this user and it's not a optimistic response.
```
Comment (luacmartins) [inline src/libs/reportUtils.js:50]:
```suggestion
 * the author is this user and it's not an optimistic response.
```

File: src/libs/reportUtils.js [hunk 3]
```diff
@@ -61,6 +62,22 @@ function canEditReportAction(reportAction) {
         && !isReportMessageAttachment(lodashGet(reportAction, ['message', 0, 'text'], ''));
 }
 
+/**
+ * Can only delete if it's a ADDCOMMENT, the author is this user and it's not a optimistic response.
```
Comment (luacmartins) [inline src/libs/reportUtils.js:66]:
```suggestion
 * Can only delete if it's an ADDCOMMENT, the author is this user and it's not an optimistic response.
```

**Review by luacmartins — CHANGES_REQUESTED** [review]
Great work! Only minor NAB changes.

**Comment by rdjuric** [comment]
Done! Thanks for the corrections @luacmartins

**Review by luacmartins — APPROVED** [review]

**Comment by roryabraham** [comment]
Unfortunately the deploy comments are not working right now. This was deployed to staging yesterday.

**Pull request merged**